### PR TITLE
Backport readmeupdate27

### PR DESCRIPTION
**Pull request by susesgartner**
Backport of:  https://github.com/rancher/rancher/pull/44601

### CODE REVIEW

File: tests/v2/validation/provisioning/hostnametruncation/README.md [hunk 1]
```diff
@@ -12,43 +12,44 @@ Please note that the config can be in either JSON or YAML (all examples are illu
...
+  rke2KubernetesVersion: ["v1.27.6+rke2r1"] #required (at least 1)
+  cni: ["calico"]                           #required (at least 1)
+  providers: ["aws"]                        #required (at least 1)
+
+awsCredentials:                            
+  secretKey: "",                            #required                                               
+  accessKey: "",                            #required                          
+  defaultRegion: ""                         #required                      
+
+awsMachineConfig:
```
Comment (slickwarren) [inline tests/v2/validation/provisioning/hostnametruncation/README.md:40]: this is now 
```
awsMachineConfigs:
  - awsMachineConfig:...
```

File: tests/v2/validation/provisioning/k3s/README.md [hunk 2]
```diff
@@ -8,118 +8,129 @@ Please see below for more details for your config. Please see below for more det
...
+provisioningInput is needed to the run the K3S tests, specifically kubernetesVersion and providers.
 
 **nodeProviders is only needed for custom cluster tests; the framework only supports custom clusters through aws/ec2 instances.**
 ```yaml
 provisioningInput:
   machinePools:
-  - nodeRoles:
-      etcd: true
+  - nodeRoles:                                              #required(dynamic only) (at least 1)
+      etcd: true                                            #required(dynamic only) (at least 1 controlplane & etcd & worker)
```
Comment (slickwarren) [inline tests/v2/validation/provisioning/k3s/README.md:38]: nit; I find the comment a little confusing but I can't think of a way to make it better 😓

File: tests/v2/validation/provisioning/k3s/README.md [hunk 3]
```diff
@@ -8,118 +8,129 @@ Please see below for more details for your config. Please see below for more det
...
       controlplane: true
       worker: true
       quantity: 1
-  - nodeRoles:
+  - nodeRoles:
```
Comment (slickwarren) [inline tests/v2/validation/provisioning/k3s/README.md:42]: nit, is this whitespace added?

File: tests/v2/validation/provisioning/k3s/README.md [hunk 4]
```diff
@@ -8,118 +8,129 @@ Please see below for more details for your config. Please see below for more det
...
       controlplane: true
       worker: true
       quantity: 1
-  - nodeRoles:
+  - nodeRoles:                  
       worker: true
       quantity: 2
       drainBeforeDelete: true
   - nodeRoles:
+      worker: true
```
Comment (slickwarren) [inline tests/v2/validation/provisioning/k3s/README.md:47]: this is not correct for windows roles, windows is its own role

Comment (slickwarren) [reply tests/v2/validation/provisioning/k3s/README.md:47]: also windows isn't supported for k3s

File: tests/v2/validation/provisioning/k3s/README.md [hunk 5]
```diff
@@ -8,118 +8,129 @@ Please see below for more details for your config. Please see below for more det
...
       windows: true
       quantity: 1
-  flags:
-    desiredflags: "Long" #These flags are for running TestProvisioningK3SCluster or TestProvisioningK3SCustomCluster it is not needed for the dynamic tests. Long will run the full table, where as short will run the short version of this test.
-  rke2KubernetesVersion: ["v1.26.8+k3s1"]
-  providers: ["linode", "aws", "do", "harvester"]
-  nodeProviders: ["ec2"]
+  rke2KubernetesVersion: ["v1.27.6+k3s1"]                    #required (at least 1)                                             
+  providers: ["linode", "aws", "do", "harvester"]            #required (at least 1)
```
Comment (slickwarren) [inline tests/v2/validation/provisioning/k3s/README.md:51]: is this list supposed to be complete? If so, we should add azure, vsphere, google, maybe others if I missed any.

**Review by slickwarren — COMMENTED** [review]
had to pause but will finish review later

File: tests/v2/validation/provisioning/k3s/README.md (hunk 5 above)
Comment (markusewalker) [reply tests/v2/validation/provisioning/k3s/README.md:51]: I would actually put the ones we mainly use. i.e. linode, aws, azure, vsphere.

**Review by markusewalker — COMMENTED** [review]
Caleb covered most if not all the nits that I had.

File: tests/v2/validation/provisioning/hostnametruncation/README.md (hunk 1 above)
Comment (susesgartner) [reply tests/v2/validation/provisioning/hostnametruncation/README.md:40]: I just rebased again and picked that up.

File: tests/v2/validation/provisioning/k3s/README.md (hunk 2 above)
Comment (susesgartner) [reply tests/v2/validation/provisioning/k3s/README.md]: Agreed it isn't the most clear. But as you said I don't know how to make it much clearer without a full explination.

File: tests/v2/validation/provisioning/k3s/README.md (hunk 3 above)
Comment (susesgartner) [reply tests/v2/validation/provisioning/k3s/README.md]: Not really, I will remove it.

**Review by markusewalker — APPROVED** [review]

**Review by markusewalker — APPROVED** [review]
LGTM

File: tests/v2/validation/provisioning/hostnametruncation/README.md (hunk 1 above)
Comment (slickwarren) [reply tests/v2/validation/provisioning/hostnametruncation/README.md:40]: looks like it isn't picket up?

**Review by Josh-Diamond — APPROVED** [review]

**Pull request merged**